link/kprobe.go

```diff
@@ -0,0 +1,233 @@
+package link
+
+import (
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/cilium/ebpf"
+)
+
+var (
+	kprobeEventsPath = filepath.Join(tracefsPath, "kprobe_events")
+
+	// Generate a unique string for each runtime instance of the program.
+	// This value is used as the 'group' token in tracefs entries to make sure
+	// multiple instances of the program don't risk clobbering each others'
+	// kprobe_events entries.
+	traceGroup = mustRandomGroup("ebpf")
+)
+
+// Kprobe returns a TraceEvent that fires when the given kernel symbol starts
+// executing. See /proc/kallsyms for available symbols. For example, printk():
+//
+//	Kprobe("printk")
+//
+// As of kernel 4.15, the presence of the given symbol is only checked when
+// attaching a program. One TraceEvent can be re-used to attach an arbitrary
+// number of programs.
+func Kprobe(symbol string) (*traceEvent, error) {
```

Comment:
I was thinking you might say the opposite. This is what I meant by `evaluate if the package is fit for external consumption`, haven't tried to consume it from outside the package. I wanted to prevent a user from instantiating a `TraceEvent{}` manually, so we'd need to add a few checks in methods on `TraceEvent`. Any tips?

Reply:
A user can always do `new(Type)` and shoot themselves in the foot. I don't consider that something to worry about.